sql/index_builder.go

```diff
@@ -67,6 +95,31 @@ func (b *IndexBuilder) Equals(ctx *Context, colExpr string, keys ...interface{})
 	}
 	potentialRanges := make([]RangeColumnExpr, len(keys))
 	for i, key := range keys {
+		// if converting from float to int results in rounding, then it's empty range
```

Comment:
Seems like there is some duplication here that might be easy to squash. What if there was a function that did a boilerplate, and you added a callback with the signature `func(key, lower, upper, err)`(no return args), put the rest of the standardized changes in the callback wrapper